db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -0,0 +1,188 @@
+  $evm.log(:info, "set requested value: prov_option: #{prov_option} value: #{option_value}")
+  return if cloud_value(prov_option, option_value, find_id, dialog_array)
+
+  $evm.log(:info, "set requested value default_option: prov_option: #{prov_option} value: #{option_value}")
+  default_option(option_value, dialog_array)
+end
+
+def cloud_value(prov_option, option_value, find_id, dialog_array)
+  return nil unless @cloud
+
+  case prov_option
+  when :cores_per_socket
+    $evm.log(:info, "set requested value cores_per_socket: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_cores_per_socket(find_id, dialog_array)
+  when :vm_memory
+    $evm.log(:info, "set requested value vm_memory: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_vm_memory(find_id, dialog_array)
+  when :storage
+    $evm.log(:info, "set requested value storage: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_storage(@miq_request.get_option(:src_vm_id), dialog_array)
+  end
+  option_set
+end
+
+def default_option(option_value, options_array)
+  return if option_value.blank?
+  options_array << option_value.to_i
+end
+
+def service_options
+  options_hash = get_dialog_options_hash(@miq_request.options[:dialog])
+  @service_template = $evm.vmdb(@miq_request.source_type, @miq_request.source_id)
+  $evm.log(:info, "service_template id: #{@service_template.id} service_type: #{@service_template.service_type}")
+  $evm.log(:info, "services: #{@service_template.service_resources.count}")
+  options_hash
+end
+
+def composite_service_options_value(child_service_resource, prov_option, options_array)
+  return if child_service_resource.resource.prov_type == 'generic'
+  child_service_resource.resource.service_resources.each do |grandchild_service_template_service_resource|
+    options_value(prov_option, grandchild_service_template_service_resource.resource, options_array)
+  end
+end
+
+# get_dialog_options_hash - Look for dialog variables in the dialog options hash that start with "dialog_option_[0-9]"
+def get_dialog_options_hash(dialog_options)
+  options_hash = Hash.new { |h, k| h[k] = {} }
+  # Loop through all of the options and build an options_hash from them
+  dialog_options.each do |k, v|
+    if /^dialog_option_(?<sequence_id>\d*)_(?<option_key>.*)/i =~ k
+      set_hash_value(sequence_id, option_key.downcase.to_sym, v, options_hash)
+    else
+      set_hash_value(0, k.downcase.to_sym, v, options_hash)
+    end
+  end
+  $evm.log(:info, "Inspecting options_hash: #{options_hash.inspect}")
+  options_hash
+end
+
+def set_hash_value(sequence_id, option_key, value, options_hash)
+  return if value.blank?
+  $evm.log(:info, "Adding seq_id: #{sequence_id} key: #{option_key.inspect} value: #{value.inspect} to options_hash")
+  options_hash[sequence_id][option_key] = value
+end
+
+def error(type)
+  msg = "Unable to calculate requested due to an error getting the #{type}"
```

Comment:
@tinaafitz 
The message looks incomplete. can it be something like
"Unable to calculate requested #{type}, due to an error getting the #{type}"